include/dlaf/multiplication/hermitian.h

```diff
@@ -21,6 +21,48 @@
 
 namespace dlaf {
 
+namespace internal {
+
+using matrix::internal::MatrixRef;
+
+template <Backend B, Device D, class T>
+void hermitian_multiplication(comm::CommunicatorGrid& grid, blas::Side side, blas::Uplo uplo,
+                              const T alpha, Matrix<const T, D>& mat_a, MatrixRef<const T, D>& mat_b,
+                              const T beta, Matrix<T, D>& mat_c) {
+  DLAF_ASSERT(matrix::square_size(mat_a), mat_a);
+  DLAF_ASSERT(matrix::square_blocksize(mat_a), mat_a);
+  DLAF_ASSERT(matrix::single_tile_per_block(mat_a), mat_a);
+  DLAF_ASSERT(matrix::single_tile_per_block(mat_b), mat_b);
+  DLAF_ASSERT(matrix::single_tile_per_block(mat_c), mat_c);
+  DLAF_ASSERT(matrix::equal_process_grid(mat_a, grid), mat_a, grid);
+  DLAF_ASSERT(matrix::equal_process_grid(mat_b, grid), mat_b, grid);
+  DLAF_ASSERT(matrix::equal_process_grid(mat_c, grid), mat_c, grid);
+
+  if (side == blas::Side::Left) {
+    DLAF_ASSERT(matrix::multipliable(mat_a, mat_b, mat_c, blas::Op::NoTrans, blas::Op::NoTrans), mat_a,
+                mat_b, mat_c);
+    switch (uplo) {
+      case blas::Uplo::Lower:
+        return multiplication::internal::Hermitian<B, D, T>::call_LL(grid, alpha, mat_a, mat_b, beta,
+                                                                     mat_c);
+        break;
+      case blas::Uplo::Upper:
+        DLAF_UNIMPLEMENTED(uplo);
+        break;
+      case blas::Uplo::General:
+        DLAF_UNIMPLEMENTED(uplo);
+        break;
+    }
+  }
+  else {
+    DLAF_ASSERT(matrix::multipliable(mat_b, mat_a, mat_c, blas::Op::NoTrans, blas::Op::NoTrans), mat_a,
+                mat_b, mat_c);
+    DLAF_UNIMPLEMENTED(side);
+  }
+}
+
+}
```

Comment:
Internally, I changed everything to use a `MatrixRef` but only for the matrix we are interested in. Therefore, this is the only function that is available for `MatrixRef` and it is called by the version using `Matrix` (so there is no code duplication). I'm fine with supporting all the different combinations of `Matrix`/`MatrixRef` but it seemed a bit overkill (it will add a combinatorial number of tests).

Reply:
After discussing in person with @rasolca yesterday, we agreed that this is probably fine since `MatrixRef` is in the `internal` namespace and all the changes are internal too. Essentially, I only modified what is needed in the internal namespace, while leaving the exposed API (with `Matrix`) unchanged. Templating over matrices will force us to test all different combinations of `Matrix`/`MatrixRef`, which is not ideal.